Bug: replication fails due to timeout and incomplete file on receiver

### What happened?

When replicating large files (~1.7GB each) from Device A to Device B, replication fails due to a timeout error. This leaves Device B with a partially written file.

### Version

v1.14.4

### OS

Linux

### Snippet to repeat the bug

Replicate large files from A to B on a local network over WiFi.

### Logs

```Shell
Device A log:
[ERROR] -- replication/remote_bucket/states/bucket_available.rs:38 Failed to write record to remote bucket http://192.168.178.54:8383/pi_robot: [Timeout] error sending request for url (http://192.168.178.54:8383/api/v1/b/pi_robot/rosbags/batch)

Device B log:
[ERROR] -- storage/entry/io/record_writer.rs:176 Failed to write record pi_robot/rosbags/1743251354672222: [BadRequest] Content is smaller than in content-length
```

### Code of Conduct

- [x] I agree to follow this project's Code of Conduct

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -7,6 +7,10 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 ## [Unreleased]
 
+### Fixed
+
+- RS-659: Fix replication timeout for large records, [PR-774](https://github.com/reductstore/reductstore/pull/774)
+
 ## [1.14.4] - 2025-03-28
 
 ### Fixed

--- a/reductstore/src/api/entry/read_query_post.rs
+++ b/reductstore/src/api/entry/read_query_post.rs
@@ -19,12 +19,6 @@ pub(crate) async fn read_query_json(
     request: QueryEntry,
     headers: HeaderMap,
 ) -> Result<QueryInfoAxum, HttpError> {
-    assert_eq!(
-        request.query_type,
-        QueryType::Query,
-        "Query type must be Query"
-    );
-
     let bucket_name = path.get("bucket_name").unwrap();
     let entry_name = path.get("entry_name").unwrap();
 

--- a/reductstore/src/replication/remote_bucket/client_wrapper.rs
+++ b/reductstore/src/replication/remote_bucket/client_wrapper.rs
@@ -66,7 +66,7 @@ impl ReductClient {
 
         let client = Client::builder()
             .default_headers(headers)
-            .timeout(std::time::Duration::from_secs(10))
+            .connect_timeout(std::time::Duration::from_secs(10))
             .danger_accept_invalid_certs(true)
             .http1_only()
             .build()

